### PR TITLE
(PA-1925) Add hiera-eyaml rubygem

### PR DESCRIPTION
**Pull request by ekinanp**
This commit also adds highline and trollop, both of which are
dependencies of hiera-eyaml.

### CODE REVIEW

**Comment by ekinanp** [comment]
Some questions:

(1) Should I go ahead and add hiera-eyaml to the agent-runtime-master project, or is that something that should be done once this gets merged in and 5.5.x CI passes?

(2) What would be some good platforms to smoke test this on? I have done it on el and ubuntu.

File: configs/projects/agent-runtime-5.5.x.rb [hunk 1]
```diff
@@ -8,4 +8,7 @@
   # Dependencies specific to the 5.5.x branch
   proj.component 'rubygem-gettext-setup'
   proj.component 'rubygem-multi_json'
+  proj.component 'rubygem-highline'
+  proj.component 'rubygem-trollop'
+  proj.component 'rubygem-hiera-eyaml'
```
Comment (caseywilliams) [inline configs/projects/agent-runtime-5.5.x.rb]: Should this go in the master runtime too?

Comment (ekinanp) [reply configs/projects/agent-runtime-5.5.x.rb]: Yeah I wasn't sure about that, asked about it just now.

Comment (caseywilliams) [reply configs/projects/agent-runtime-5.5.x.rb]: Typically we've been just applying component changes like this to all the branches they're supposed to be in at once -- but on the other hand, until this point the work we've been doing in this repo has been focused on migrating things that used to be in puppet-agent to this repo. That said, if everything builds and you can get some agent acceptance tests to pass, I wouldn't feel bad about adding these gems to both 5.5.x and master if no one else has a strong opinion.

**Comment by branan** [comment]
1.  adding to master is fine. I don't want us to be trying to manually "merge up" changes in the runtime. That's too much effort and possibility for confusion or missed work.
2. I'd like to see it on at least one of the "weird" platforms (solaris or AIX) and a windows, pluse the EL and Ubuntu you've already got

**Comment by ekinanp** [comment]
Tests passed. Here are some relevant links:

(Runtime build)
https://jenkins-master-prod-1.delivery.puppetlabs.net/view/__experimental/job/experimental_agent-runtime-composite_runtime-vanagon-packaging_adhoc_5.5.x/ 

(Agent build)
https://jenkins-master-prod-1.delivery.puppetlabs.net/view/__experimental/job/experimental_agent-runtime-composite_puppet-agent-packaging_adhoc_5.5.x/

(Agent integration tests)
https://jenkins-master-prod-1.delivery.puppetlabs.net/view/__experimental/job/experimental_agent-runtime-composite_puppet-agent-integration-suite_adhoc_5.5.x/

(Verify that my 5.5.x agent uses this runtime build)
https://github.com/ekinanp/puppet-agent/blob/5.5.x/configs/components/puppet-runtime.json

(Verify that the runtime tag corresponds to the PR commit)
https://github.com/ekinanp/puppet-runtime/tree/201804100

I also downloaded the built agent package on an EL 7 machine and verified that the three gems added here are present in the shared gem directory.

So this should be good to go

**Pull request merged**